### ISSUE TITLE
[EX-52] Add a "failed" feature to mark tasks that couldn't be completed successfully

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#48]
_Michele Santullo on 2013-08-29T09:25:41Z says:_ 

 It would be nice to have a "failed" feature, as well as "canceled" possibly, to better log if and why a task was completed. "done" means it was completed, and as discussed on IRC annotations could do the job, but consider the following example:
I have to take a train to the beach tomorrow at 10:00 (task 1), then I have to go sunbathing at 16:00 (task 2) and call my friend there at 18:00 to agree on a bar we should meet at (task 3). Tasks 2 and 3 depend on 1. Sadly, I miss the train, so I'm unable to complete task 2 and it should be automatically marked as "canceled due to failed blocking task". However, task 3 should still be unblocked, as I still have to call my friend to tell him we can't meet anymore. I suggest something like this:

> add take the train due:20130829-1000
> add go sunbathing due:20130839-1600 depends:1s
> add call my friend and agree on a bar to meet at depends:1 due:20130829-1800

The "s" in "depends:1s" stands for "strong", so taskwarrior would know that task 2 can't be done unless task 1 is completed successfully. Then as I miss the train I do:

> 1 failed

and I expect task 2 to be canceled with a reason "can't be done due to blocking task failure", and task 3 to be unblocked as usual. When I look for the completed tasks, it would be useful to clearly see at a glance that task 1 failed, task 2 was canceled and task 3 was done. Also, if possible I wouldn't abuse the annotations as they can become quite much in number, so while an annotation "failed" could do the trick somehow (or a tag even), it could still happen that you add annotations after a task was completed and the "failed" part would get lost in the lot.
Adding a reason to "done" could do the trick. One could mark a task as failed through "1 done failed", done with "1 done" and canceled with "1 done canceled", and even add a custom reason with "1 done reason my co-worker did it for me". I'm not sure custom reasons should be allowed, but it could be considered.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2013-08-29T09:25:41Z
Modified: 2017-01-17T02:42:51Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2013-08-29T15:04:27Z says:_ 

 <pre>
1 done failed
</pre>

This is already supported.  This command:

<pre>
task 1 done failed
</pre>

Will mark task 1 as completed, and add the annotation 'failed' to the task.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2013-09-29T16:24:09Z says:_ 

 Fundamentally, a task is either completed or it isn't.  This is reflected in the "completed" and "deleted" status.  Any subtle refinements to those meanings belong in a UDA field, or annotation.  Status is not likely to be expanded.

When dependencies were designed, we discussed whether the dependency should be enforced, or whether the user should have the freedom to do anything.  We chose freedom, with warnings.  Nothing is enforced.

The chaining effect you describe (delete 1, then 2 is deleted also) is something that we would implement in an external script.  I'm reassigning this to "extension", because the idea has merit, and people might want this, but it is not core functionality.
